### DISCUSSION TITLE
return values from tasks sent to goChron

### DISCUSSION THREAD

**Discussion by avimess23** [go-co-op/gocron#344] [General]
Hi all,
I would like to use goChron for my program but I cant seem to find a built in way to retrieve the return values from the function I am schedualing.
I have concidered using a channel, which would solve the problem, but I wanted to make sure it was absolutely necessary before I change my function's code.

Is there already a way to retrieve said values from the scheduled function? 
Thanks

**Comment by JohnRoesler** [comment]
@avimess23 Today this isn't an option. You would have to add a channel in your function to send values back on.

I have considered adding support for this as the `Call` method https://github.com/go-co-op/gocron/blob/main/gocron.go#L93 does return a slice of `[]reflect.Values` which could be returned on channel. However, the user would then have to handle getting the underlying type of the data. Probably easier to just keep it such that if a user cares about the return values, to instead pass them back on a channel and handle as they see fit.

**Reply by avimess23** [reply]
Thanks so much for your reply!
I understand the trade off. 
Thanks
